.github/workflows/ssp-ci.yml

```diff
@@ -0,0 +1,32 @@
+name: Self-serve Portal CI
+on: workflow_dispatch
+
+jobs:
+  check-email-templates:
+    runs-on: ubuntu-latest
+    name: Check email template updates
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Get all email templates that have changed
+        id: changed-files
+        uses: tj-actions/changed-files@v39
+        with:
+          files_yaml: |
+            emailTemplates:
+              - 'emailTemplates/**.hbs'
+    outputs:
+      emailTemplatesUpdated: ${{ steps.changed-files.emailTemplates_any_changed }}
+  sync-email-templates:
+    needs: [check-email-templates]
+    name: Sync handlebars template with SendGrid
+    if: needs.check-email-templates.outputs.emailTemplatesUpdated == 'true'
+    uses: ./.github/workflows/sync-sendgrid-template-action.yml
+    with:
+      FORCE_SYNC_ALL: 'true'
+    secrets:
+      SENDGRID_API_KEY: ${{ secrets.SENDGRID_API_KEY }}
+  build-and-publish-ssportal:
+    needs: [sync-email-templates]
+    if: always() && (needs.sync-email-templates.result == 'success' || needs.sync-email-templates.result == 'skipped') # Run build and publish ssportal when sync email template is ran or skipped
```

Comment:
what's the point of this conditional if there is an `always()`? won't it just run every time anyway?

Reply:
Coz I use needs here to make sure `build-and-publish-ssportal` runs after `sync-email-templates`, without always(), `build and publish ssportal` will be skipped if sync is skipped

Reply:
okay - so does that mean that `needs.sync-email-templates.result == 'skipped'` does not cover the case if the sync is skipped? I still can't see the point of `(needs.sync-email-templates.result == 'success' || needs.sync-email-templates.result == 'skipped')` if we have an `always` - I must be missing something

Reply:
so I use `always()` to make sure `build-and-publish-ssportal`should always run after there is result of sync-email-templates. And Use `needs.sync-email-templates.result == 'success' || needs.sync-email-templates.result == 'skipped'` to make sure `build-and-publish-ssportal` should not run if sync-email-templates is failed

Reply:
Without always(), both skipped and failed will skip the `build-and-publish-ssportal` and without the result check, build and publish ssportal will still run after sync-email-templates failed

Reply:
ahh in my head I was reading it as `always() ||` not `always() &&` for some reason 🤦  thanks for explaining!